Backend/index.js

```diff
@@ -1,22 +1,49 @@
 const express = require('express');
 const app = express();
 
+const flash = require('express-flash'); 
+const passport = require('passport');
+const session = require('express-session');
+const methodOverride = require('method-override');
+const bodyParser = require('body-parser');
+app.use(bodyParser.urlencoded({ extended: true }));
+
+//Initializes Passport
+const initializePassport = require('./passport/passport-config');
+const getUserById = require('./databaseInteraction/getUserById');
+initializePassport(
+  passport,
+  getUserById
+); 
+
 //For if a port is supplied as an enviroment variable
 const port = process.env.PORT || 3001;
 
 app.use(express.static('public'));
 app.use(express.json());
-
-app.listen(port, () => {
-  console.log(`Listening at http://localhost:${port}`);
-});
+app.use(flash());
+app.use(session({
+  secret: 'secret',
+  resave: false,
+  saveUninitialized: false
+}));
+app.use(passport.initialize());
+app.use(passport.session());
+app.use(methodOverride('_method'));
 
 const endpoints = [
   require('./endpoints/testGet'),
-  require('./endpoints/loginPost'),
+  require('./endpoints/getUser'),
+  require('./endpoints/logout'),
   require('./endpoints/listingsPost')
 ];
 
 for(const endpoint of endpoints){
   endpoint(app);
-}
\ No newline at end of file
+}
+
+require('./endpoints/loginPost')(app, passport);
```

Comment:
It might be nice to do all the passport set up in a different file (in the `Backend/passport` directory for example) to keep the `index.js` file as clean as possible. This would have the advantage of compartmentalizing the passport stuff, and since it would only require the express app (for the `app.use` calls) it could fit with the other endpoints. Currently, this works so no change is needed.